### PR TITLE
optimize make

### PR DESCRIPTION
**Pull request by daixiang0**
Optimize make:

- unite binary build
- add clean target
- update test target to sync with CI
- use lint target in CI to ensure sync

Signed-off-by: Loong Dai <loong.dai@intel.com>

### CODE REVIEW

File: Makefile.targets.mk [hunk 1]
```diff
@@ -12,24 +12,26 @@ REV=$(shell git rev-parse --short HEAD)
...
+SUPPORT_ARCHS ?= amd64 arm64
+# Generate three targets: build-amd64, build-arm64
+# Run make build -n to see the result with dry run.
+BUILD_BINARY_ARCHS = $(addprefix build-,$(SUPPORT_ARCHS))
 
-build-linux-amd64:
-	@GOOS=linux GOARCH=amd64 $(MAKE) build
+.PHONY: build $(BUILD_BINARY_ARCHS)
+build: $(BUILD_BINARY_ARCHS) ## Build the envoy-gateway binary.
+$(BUILD_BINARY_ARCHS): build-%:
```
Comment (arkodg) [inline Makefile.targets.mk:22]: this breaks `darwin` binaries

Comment (daixiang0) [reply Makefile.targets.mk:22]: Now `make build` replaces `make build-all` as `build-all` only support Linux, I will update for darwin,

Comment (arkodg) [reply Makefile.targets.mk:22]: you might end up building something similar to what @Xunzhuo  already did in https://github.com/envoyproxy/gateway/pull/113, lets wait for @Xunzhuo to wrap up https://github.com/envoyproxy/gateway/pull/113 and then see if any more optimizations are needed.
thanks !

Comment (daixiang0) [reply Makefile.targets.mk:26]: Now:

```
root:[gateway]$ MAKE_IN_DOCKER=0 make build -n
make -f Makefile.targets.mk build
make[1]: Entering directory '/root/github/gateway'
CGO_ENABLED=0 GOOS="linux" GOARCH="amd64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
CGO_ENABLED=0 GOOS="linux" GOARCH="arm64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
CGO_ENABLED=0 GOOS="darwin" GOARCH="amd64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
CGO_ENABLED=0 GOOS="darwin" GOARCH="arm64" go build -a -o ./bin/linux/amd64/ github.com/envoyproxy/gateway/cmd/envoy-gateway
make[1]: Leaving directory '/root/github/gateway'
```

Comment (daixiang0) [reply Makefile.targets.mk:26]: > you might end up building something similar to what @Xunzhuo already did in #113, lets wait for @Xunzhuo to wrap up #113 and then see if any more optimizations are needed. thanks !

Sure, I can wait.

Comment (daixiang0) [reply Makefile.targets.mk:62]: @arkodg seems my implementation looks easier to maintain, please take a look.

**Pull request closed without merging**